src/validation.cpp

```diff
@@ -1851,6 +1851,18 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         flags |= SCRIPT_VERIFY_NULLDUMMY;
     }
 
+    // SEGWIT2X signalling.
+    if ( VersionBitsState(pindex->pprev, chainparams.GetConsensus(), Consensus::DEPLOYMENT_SEGWIT2X, versionbitscache) == THRESHOLD_ACTIVE &&
```

Comment:
> I might want to change my mind when it turns out 80% does, in which case I don't want to be orphaned. To achieve this, a LOCKED_IN period is required, and 2 weeks is reasonable.

One thought I had is that since BIP141 will lock in after signaling 1916 out of the 2016 blocks in a difficulty readjustment period, it would be possible to start enforcing during the LOCKED_IN state *and* give miners a short upgrade period by delaying the enforcement by ~100 blocks, and still guarantee BIP141 lock-in by the next readjustment. So something like (state == ACTIVE || (state == LOCKED_IN && blockheight % 2016 >= 100).

That would of course only give a grace period of about 16 hours depending on block luck, rather than the approximately two weeks. I expect miners will be watching this closely, so this may be sufficient, but again, is of course still riskier orphaning-wise than waiting a full period.